CustomArtifact builder panics if no dependencies specified

We should fail a bit more gracefully and/or have meaningful defaults. 

````
$ skaffold dev
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x127eaa6]

goroutine 1 [running]:
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.validateCustomDependencies(0xc0005bc088,
0x1, 0x1, 0x0, 0x0, 0x0)
/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:72
+0x66
github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation.Process(0xc0005d8340,
0x0, 0x0)
/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation/validation.go:37
+0x161
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.newRunner(0x2515e40,
0xc0005dc1e0, 0xc00029bc00, 0xc000594080, 0xc000590060)
/go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/runner.go:60
+0xc7
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.dev(0x17bbb20,
0xc00000c018, 0x0, 0x0)
/go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/dev.go:73
+0x149
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.NewCmdDev.func1(0xc0003fb900,
0x25347f8, 0x0, 0x0, 0x0, 0x0)
/go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/dev.go:36
+0x33
github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra.(*Command).execute(0xc0003fb900,
0x25347f8, 0x0, 0x0, 0xc0003fb900, 0x25347f8)
/go/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra/command.go:762
+0x473
github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc0003fa280,
0xc0003fb900, 0xc0003fb680, 0xc0003fb400)
/go/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra/command.go:852
+0x2fd
github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra.(*Command).Execute(0xc0003fa280,
0xc00000c018, 0x17bbb20)
/go/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra/command.go:800
+0x2b
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app.Run(0x17bbb20,
0xc00000c018, 0x17bbb20, 0xc00000c020, 0xc0000ac058, 0x0)
/go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/skaffold.go:27
+0x57
main.main()
/go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/skaffold.go:30
+0x4e
````

--- a/pkg/skaffold/schema/validation/validation.go
+++ b/pkg/skaffold/schema/validation/validation.go
@@ -67,12 +67,10 @@ func validateDockerNetworkMode(artifacts []*latest.Artifact) (errs []error) {
 // validateCustomDependencies makes sure that dependencies.ignore is only used in conjunction with dependencies.paths
 func validateCustomDependencies(artifacts []*latest.Artifact) (errs []error) {
 	for _, a := range artifacts {
-		if a.CustomArtifact == nil {
+		if a.CustomArtifact == nil || a.CustomArtifact.Dependencies == nil || a.CustomArtifact.Dependencies.Ignore == nil {
 			continue
 		}
-		if a.CustomArtifact.Dependencies.Ignore == nil {
-			continue
-		}
+
 		if a.CustomArtifact.Dependencies.Dockerfile != nil || a.CustomArtifact.Dependencies.Command != "" {
 			errs = append(errs, fmt.Errorf("artifact %s has invalid dependencies; dependencies.ignore can only be used in conjunction with dependencies.paths", a.ImageName))
 		}

--- a/pkg/skaffold/schema/validation/validation_test.go
+++ b/pkg/skaffold/schema/validation/validation_test.go
@@ -500,6 +500,9 @@ func TestValidateCustomDependencies(t *testing.T) {
 				Ignore:  []string{"ignoreme"},
 			},
 			expectedErrors: 1,
+		}, {
+			description:  "nil dependencies",
+			dependencies: nil,
 		},
 	}
 	for _, test := range tests {

